src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -635,6 +635,13 @@ def what_if_deploy_arm_template_at_subscription_scope(cmd,
 def _what_if_deploy_arm_template_core(cli_ctx, what_if_poller, no_pretty_print):
     what_if_result = LongRunningOperation(cli_ctx)(what_if_poller)
 
+    if what_if_result.error:
+        # The status code is 200 even when there's an error, because
+        # it is technically a successful What-If operation. The error
+        # is on the ARM template but not the operation.
+        err_message = _build_what_if_error_message(what_if_result.error)
+        raise CLIError(err_message)
+
     if no_pretty_print:
         return what_if_result
```

Comment:
What about the error in the `what_if_poller` operation itself? How is this situation handled? Will exceptions be thrown?

Reply:
Yes, exceptions will be thrown and handled by `handle_template_based_exception` specified in the command definitions.